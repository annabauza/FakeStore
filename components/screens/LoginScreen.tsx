import React, { useContext, useState } from 'react';
import { View, Button, TextInput, StyleSheet } from 'react-native';
import { NavigationScreenProp } from 'react-navigation';
import UserContext from '../../contexts/UserContext';
import { loginUser } from '../../network/API';
import { ErrorHandler } from './BaseScreen';

export interface LoginScreenProps {
  navigation: NavigationScreenProp<any, any>;
}

const LoginScreen: React.FC<LoginScreenProps> = ({ navigation }) => {
  const { setIsLoggedIn, setToken, setUsername } = useContext(UserContext);
  const [state, setState] = useState('loaded');
  const [login, setLogin] = useState('mor_2314');
  const [errorMessage, setErrorMessage] = useState(
    'Could not login, please try again later',
  );
  const [password, setPassword] = useState('83r5');

  return ErrorHandler(
    state,
    errorMessage,
    <View>
      <TextInput
        style={styles.input}
        onChangeText={txt => {
          setLogin(txt);
        }}
        value={login}
      />
      <TextInput
        style={styles.input}
        onChangeText={txt => {
          setPassword(txt);
        }}
        value={password}
      />
      <Button
        title={'login'}
        onPress={() => {
          setState('loading');
          loginUser(
            login,
            password,
            token => {
              console.log({ setUsername, login });
              setToken ? setToken(token) : null;
              setUsername ? setUsername(login) : null;
              setState('loaded');
              setIsLoggedIn ? setIsLoggedIn(true) : null;
              navigation.goBack('Home');
            },
            error => {
              setErrorMessage(error.message);
              setState('error');
              console.error(error);
            },
          );
        }}
      />
    </View>,
    () => {
      setState('loaded');
    },
  );
};

const styles = StyleSheet.create({
  input: {
    height: 40,
    margin: 12,
    borderWidth: 1,
    padding: 10,
  },
});

export default LoginScreen;
