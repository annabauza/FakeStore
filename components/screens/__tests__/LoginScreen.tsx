import React from 'react';
import renderer from 'react-test-renderer';
import LoginScreen from '../LoginScreen';

test('renders correctly', async () => {
  const tree = renderer.create(<LoginScreen navigation={undefined} />).toJSON();
  expect(tree).toMatchSnapshot();
});
