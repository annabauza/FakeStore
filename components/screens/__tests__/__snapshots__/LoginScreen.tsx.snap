// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <TextInput
    onChangeText={[Function]}
    style={
      Object {
        "borderWidth": 1,
        "height": 40,
        "margin": 12,
        "padding": 10,
      }
    }
    value="mor_2314"
  />
  <TextInput
    onChangeText={[Function]}
    style={
      Object {
        "borderWidth": 1,
        "height": 40,
        "margin": 12,
        "padding": 10,
      }
    }
    value="83r5"
  />
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        login
      </Text>
    </View>
  </View>
</View>
`;
